corehq/apps/data_interfaces/tests/test_case_deduplication.py

```diff
@@ -1152,10 +1194,16 @@ def test_include_closed_finds_open_and_closed_cases(self):
         self._set_up_rule(include_closed=True)
 
         backfill_deduplicate_rule(self.domain, self.rule)
-        self.assertEqual(CaseDuplicate.objects.filter(action=self.action).count(), 3)
+
+        duplicate_case_ids = CaseDuplicate.objects.filter(action=self.action).values_list('case_id', flat=True)
+        self.assertEqual(len(duplicate_case_ids), 3)
+        self.assertNotIn(self.case4.case_id, duplicate_case_ids)
```

Comment:
I'm curious why this test is verifying that the closed case was excluded? Is this already covered by `test_find_duplicates_exclude_copied_cases`?

Reply:
Assuming the query is as to why this test was updated - This test `test_include_closed_finds_open_and_closed_cases` and below one `test_finds_open_cases_only` were updated to make sure cases irrespective of open/closed are excluded if they are copied. More like an Integration test for the Deduplication feature.

Reply:
Hmm, I see. So `test_find_duplicates_exclude_copied_cases` was intended to test the elasticsearch query itself, while we're using `test_include_closed_finds_open_and_closed_cases` to verify the new logic introduced in the `when_case_matches` method?  Do you think it would make sense to add tests explicitly verifying this functionality in _data_interfaces.tests.test_case_deduplication:CaseDeduplicationActionTest_? That test suite is lower-to-the-ground, specifically testing `run_rule`, which ultimately calls `when_case_matches`. I think the backfill tests here were intended to test the backfill population, and I think the copy logic is something we'd like verified at a lower level.

That said, fully aware that the dedupe tests are kind of a tangle to wade thru -- they exist in a lot of different places.

Reply:
> Do you think it would make sense to add tests explicitly verifying this functionality in data_interfaces.tests.test_case_deduplication:CaseDeduplicationActionTest? That test suite is lower-to-the-ground, specifically testing run_rule, which ultimately calls when_case_matches

Yes on re-thinking looks like this could be a better place to test the same.